### PR TITLE
debugstatus: add Handler

### PR DESCRIPTION
**Pull request by rogpeppe**
This code is already in two services and rather than add it to a third, we factor
it out to here.

(Review request: http://reviews.vapour.ws/r/2990/)

### CODE REVIEW

File: debugstatus/handler.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+// Handler implements a type that can be used with httprequest.Handlers
+// to serve a standard set of /debug endpoints, including
+// the version of the system, its current health status
+// the runtime profiling information.
+type Handler struct {
+	// Check will be called to obtain the current health of the
+	// system. It should return a map as returned from the
+	// Check function.
+	Check func() map[string]CheckResult
```
Comment (mhilton) [inline debugstatus/handler.go]: I think the Handler shouldn't require you to set all of these things to use it. If Check of CheckPprofAllowed is nil then we should do something that isn't panicking. /debug/check could just return an empty set (not sure if that should be considered successful or not) CheckPprofAllowed could just deny access.

**Comment by mhilton** [comment]
LGTM with one thought about default behaviour

File: debugstatus/handler.go (hunk 1 above)
Comment (rogpeppe) [inline debugstatus/handler.go]: done

File: debugstatus/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+		Message: err.Error(),
+	}
+}
+
+type handlerSuite struct {
+}
+
+var _ = gc.Suite(&handlerSuite{})
+
+var errUnauthorized = errgo.New("you shall not pass!")
```
Comment (mattyw) [inline debugstatus/handler_test.go]: cute - but "unauthorized" to make the error a bit more explicit

**Comment by mattyw** [comment]
LGTM - wish I'd had this a few months ago

File: debugstatus/handler_test.go (hunk 2 above)
Comment (rogpeppe) [inline debugstatus/handler_test.go]: naa, i'm gonna go with cuteness. :)

**Comment by rogpeppe** [comment]
$$merge$$ :shipit:

**Pull request merged**